test/unit/math/prim/fun/as_value_column_vector_or_scalar_test.cpp

```diff
@@ -0,0 +1,73 @@
+#include <stan/math/prim/fun/Eigen.hpp>
+#include <stan/math/prim/fun/as_value_column_vector_or_scalar.hpp>
+#include <test/unit/util.hpp>
+#include <gtest/gtest.h>
+
+TEST(MathMetaPrim, as_value_column_vector_or_scalar_scalar) {
+  using stan::math::as_value_column_vector_or_scalar;
+  int a = 3;
+  double b = 4;
+  EXPECT_EQ(a, as_value_column_vector_or_scalar(a));
+  EXPECT_EQ(b, as_value_column_vector_or_scalar(b));
+}
+
+TEST(MathMetaPrim, as_value_column_vector_or_scalar_std_vector_lvalue) {
+  using stan::math::as_value_column_vector_or_scalar;
+  int n = 100;
+  Eigen::VectorXd a = Eigen::VectorXd::Random(n);
+  std::vector<double> b(n);
+  for (int i = 0; i < n; i++) {
+    b[i] = a.coeff(i);
+  }
+  auto&& tmp = as_value_column_vector_or_scalar(b);
+  a[0] = tmp[0] = 12345;
```

Comment:
Should this be

`a[0] = b[0] = 12345;`

I don't understand how assigning `a` and `tmp` here tests anything.

But `tmp` and `b` point at the same memory I think.